drivers/wifi/siwx917/siwx917_wifi.c

```diff
@@ -411,6 +411,62 @@ static int siwx917_ap_disable(const struct device *dev)
 	sidev->state = WIFI_STATE_INTERFACE_DISABLED;
 	return ret;
 }
+
+static int siwx917_ap_sta_disconnect(const struct device *dev, const uint8_t *mac_addr)
+{
+	sl_mac_address_t mac = {};
+	int ret;
+
+	if (!mac_addr) {
+		return -EINVAL;
+	}
+
+	strncpy(mac.octet, mac_addr, ARRAY_SIZE(mac.octet));
+
+	ret = sl_wifi_disconnect_ap_client(SL_WIFI_AP_2_4GHZ_INTERFACE, &mac, SL_WIFI_DEAUTH);
+	if (ret) {
+		LOG_ERR("Failed	to disconnect: 0x%x", ret);
+		return -EIO;
+	}
+
+	return ret;
+}
+
+static sl_status_t ap_connected_event_handler(sl_wifi_event_t event, void *data,
```

Comment:
Let's be consistent in the function name:
 - Prefix the function with  `siwx917_`
 - The other callbacks in this file are named `on_<event>` (so your function names will be shorter and easier to read)